Description: Fix PIL imports for pillow compatibility --
 see 5117D0B7.801@debian.org (LP: 1160571).
 There's a version of this patch (without the checks for ImportError) in
 upstream's patch tracker at https://sourceforge.net/p/pysolfc/patches/10/
Author: Bernhard Reiter <ockham@raz.or.at>
Forwarded: no

--- a/pysollib/mfxutil.py
+++ b/pysollib/mfxutil.py
@@ -41,18 +41,30 @@
 Image = ImageTk = ImageOps = None
 if TOOLKIT == 'tk':
     try: # PIL
-        import Image
-        import ImageTk
-        import ImageOps
+        try:
+            from PIL import Image
+            from PIL import ImageTk
+            from PIL import ImageOps
+        except ImportError:
+            import Image
+            import ImageTk
+            import ImageOps
     except ImportError:
         Image = None
     else:
         # for py2exe
-        import GifImagePlugin
-        import PngImagePlugin
-        import JpegImagePlugin
-        import BmpImagePlugin
-        import PpmImagePlugin
+        try:
+            from PIL import GifImagePlugin
+            from PIL import PngImagePlugin
+            from PIL import JpegImagePlugin
+            from PIL import BmpImagePlugin
+            from PIL import PpmImagePlugin
+        except ImportError:
+            import GifImagePlugin
+            import PngImagePlugin
+            import JpegImagePlugin
+            import BmpImagePlugin
+            import PpmImagePlugin
         Image._initialized = 2
 
 
--- a/scripts/cardset_viewer.py
+++ b/scripts/cardset_viewer.py
@@ -7,7 +7,10 @@
 from math import sqrt, sin, cos, pi
 from Tkinter import *
 try:
-    import Image, ImageTk
+    try:
+        from PIL import Image, ImageTk
+    except ImportError:
+        import Image, ImageTk
 except ImportError:
     Image = None
 
